### PR TITLE
[Customer] Changing an email of a customer

### LINKED ISSUE #8597: Change customer email doesn't change it in User entity
I try to change the email of a customer from the admin but the email property of the User entity is not changed

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

File: features/user/managing_customers/changing_customer_email.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
+@managing_customers
+Feature: Changing an email of an existing customer
+    In order to change a customer's email
```
Comment (lchrusciel) [inline features/user/managing_customers/changing_customer_email.feature]: In order to be able to help a customer when he forgets his password

File: src/Sylius/Behat/Context/Ui/Admin/ManagingCustomersContext.php [hunk 2]
```diff
@@ -116,6 +116,15 @@ public function iSpecifyItsEmailAs($email = null)
     }
 
     /**
+     * @When I change their email to :email
+     * @When I remove its email
+     */
+    public function iChangeTheirEmailTo(string $email = null): void
```
Comment (pamil) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingCustomersContext.php]: ?string

File: src/Sylius/Component/Core/Model/Customer.php [hunk 3]
```diff
@@ -54,6 +54,18 @@ public function __construct()
     /**
      * {@inheritdoc}
      */
+    public function setEmail(?string $email): void
+    {
+        parent::setEmail($email);
+
+        if ($this->hasUser() && null != $email) {
+            $this->getUser()->setUsername($email);
```
Comment (pamil) [inline src/Sylius/Component/Core/Model/Customer.php]: Shouldn't we update user's email?

**Comment by pjedrzejewski** [comment]
Thanks Grzesiu! 👍

**Pull request merged**